### PR TITLE
Fixes #4267 Add word filters pagination

### LINKED ISSUE #4267: Add word filters pagination
My forum need to add thousands of word filters, mybb version is 1.8.24.

But mybb default word filtering is not paging, which will leads to very lengthy and slow loading.

Want pagenation like this

In badwords.php

$query = $db-&gt;simple_select(&quot;badwords&quot;, &quot;*&quot;, &quot;&quot;, array(&quot;order_by&quot; =&gt; &quot;badword&quot;, &quot;order_dir&quot; =&gt; &quot;asc&quot;));
	while($badword = $db-&gt;fetch_array($query))
	{
		$badword['badword'] = htmlspecialchars_uni($badword['badword']);
		$badword['replacement'] = htmlspecialchars_uni($badword['replacement']);
		if(!$badword['replacement'])
		{
			$badword['replacement'] = '*****';
		}

		$regex = $lang-&gt;no;
		if($badword['regex'])
		{
			$regex = $lang-&gt;yes;
		}

		$table-&gt;construct_cell($badword['badword']);
		$table-&gt;construct_cell($badword['replacement']);
		$table-&gt;construct_cell($regex, array(&quot;class&quot; =&gt; &quot;align_center&quot;));
		$table-&gt;construct_cell(&quot;&lt;a href=\&quot;index.php?module=config-badwords&amp;amp;action=edit&amp;amp;bid={$badword['bid']}\&quot;&gt;{$lang-&gt;edit}&lt;/a&gt;&quot;, array(&quot;class&quot; =&gt; &quot;align_center&quot;));
		$table-&gt;construct_cell(&quot;&lt;a href=\&quot;index.php?module=config-badwords&amp;amp;action=delete&amp;amp;bid={$badword['bid']}&amp;amp;my_post_key={$mybb-&gt;post_code}\&quot; onclick=\&quot;return AdminCP.deleteConfirmation(this, '{$lang-&gt;confirm_bad_word_deletion}');\&quot;&gt;{$lang-&gt;delete}&lt;/a&gt;&quot;, array(&quot;class&quot; =&gt; &quot;align_center&quot;));
		$table-&gt;construct_row();
	}
I may need to add limit and change this query

$query = $db-&gt;simple_select(&quot;badwords&quot;, &quot;*&quot;, &quot;&quot;, array(&quot;order_by&quot; =&gt; &quot;badword&quot;, &quot;order_dir&quot; =&gt; &quot;asc&quot;));
I am a php rookie, I don’t know how to make other changes(and add query parameters in template?)  :rolleyes: 

Hope to get help, thanks in advance! :heart:

Original thread: [How to change word filters to pagination?](https://community.mybb.com/thread-231330.html)

### PR DESCRIPTION
**Pull request by Starpaul20**
Resolves #4267 

Also moved the add word filter to its own page (for consistency with the rest of the Admin CP).

### CODE REVIEW

File: admin/modules/config/badwords.php [hunk 1]
```diff
@@ -18,85 +18,119 @@
...
+	if($errors)
 	{
-		$mybb->input['action'] = '';
+		$page->output_inline_error($errors);
 	}
+
+	$form_container = new FormContainer($lang->add_bad_word);
+	$form_container->output_row($lang->bad_word." <em>*</em>", $lang->bad_word_desc, $form->generate_text_box('badword', $mybb->get_input('badword'), array('id' => 'badword')), 'badword');
+	$form_container->output_row($lang->replacement, $lang->replacement_desc, $form->generate_text_box('replacement', $mybb->get_input('replacement'), array('id' => 'replacement')), 'replacement');
+	$form_container->output_row($lang->regex, $lang->regex_desc, $form->generate_yes_no_radio('regex', $mybb->get_input('regex'), array('id' => 'regex')), 'regex');
```
Comment (Sama34) [inline admin/modules/config/badwords.php]: It should use `$mybb->get_input('regex', MyBB::INPUT_INT)`, the purpose being that enabling the regex setting on by default could cause confusion when the filter isn't an actual regex (i.e. newies or people who don't read).

@mybb/developers

**Review by Sama34 — CHANGES_REQUESTED** [review]

**Comment by Ben-MyBB** [comment]
I can’t check at the moment but Iirc this issue also applies to the IP ban list

**Comment by Starpaul20** [comment]
There's actually a lot of places in the Admin CP that could use paging (Attachment Types, Ban filters, MyCodes, Spiders, Thread Prefixes, etc.)

**Comment by euantorano** [comment]
> There's actually a lot of places in the Admin CP that could use paging (Attachment Types, Ban filters, MyCodes, Spiders, Thread Prefixes, etc.)

Agreed, but would like to see separate issues and PRs for each one to make reviewing as easy as possible.

**Review by Sama34 — APPROVED** [review]

**Comment by Eldenroot** [comment]
Tested, works fine! Thank you!

**Pull request merged**